spring-cloud-kubernetes-integration-tests/spring-cloud-kubernetes-k8s-client-discovery-server/src/main/java/org/springframework/cloud/kubernetes/discoveryclient/it/KubernetesDiscoveryClientApplicationIt.java

```diff
@@ -30,27 +32,36 @@
  * @author Ryan Baxter
  */
 @SpringBootApplication
+@EnableScheduling
 @RestController
-public class KubernetesDiscoveryClientApplicationIt {
+class KubernetesDiscoveryClientApplicationIt {
 
 	private final DiscoveryClient discoveryClient;
 
-	public KubernetesDiscoveryClientApplicationIt(DiscoveryClient discoveryClient) {
+	private final HeartbeatListener heartbeatListener;
+
+	KubernetesDiscoveryClientApplicationIt(DiscoveryClient discoveryClient, HeartbeatListener heartbeatListener) {
 		this.discoveryClient = discoveryClient;
+		this.heartbeatListener = heartbeatListener;
 	}
 
 	public static void main(String[] args) {
 		SpringApplication.run(KubernetesDiscoveryClientApplicationIt.class, args);
 	}
 
 	@GetMapping("/services")
-	public List<String> services() {
+	List<String> services() {
 		return discoveryClient.getServices();
 	}
 
 	@GetMapping("/service/{serviceId}")
-	public List<ServiceInstance> service(@PathVariable String serviceId) {
+	List<ServiceInstance> service(@PathVariable String serviceId) {
 		return discoveryClient.getInstances(serviceId);
 	}
 
+	@GetMapping("/state")
```

Comment:
a new endpoint in controller that will get the heartbeat event, this is later used in tests for proper assertions